### PR TITLE
Add ReferenceConverter entry to intrinsic type converters table

### LINKED ISSUE #34709: ReflectTypeDescriptorProvider fails to add ReferenceConverter to intrinsic converter table
`ReflectTypeDescriptorProvider` _should_ have `ReferenceConverter` in its intrinsic type converter table, but it doesn't. This has significant user impact at design-time. For example, when assigning certain properties via the `PropertyGrid`, such as `Form.AcceptButton`, the `ReferenceConverter` is used to produce a list of potential objects that can be referenced. However, in .NET Core, this doesn't work because the `ReferenceConverter` doesn't get instantiated for that case.

https://github.com/dotnet/runtime/blob/e298818e7c1327178de8168ed785c0e23d14d298/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs#L130-L136

The code above should include the following line:

```C#
[s_intrinsicReferenceKey] = typeof(ReferenceComparer),
```

The `s_intrinsicReferenceKey` is properly declared and created in `ReflectTypeDescriptor`, but it looks like it was simply missed in the table.

On .NET Framework, `ReferenceConverter` is correctly added to the table of intrinsic type converters: https://referencesource.microsoft.com/#System/compmod/system/componentmodel/ReflectTypeDescriptionProvider.cs,146

### PR DESCRIPTION
**Pull request by safern**
Fixes: https://github.com/dotnet/runtime/issues/34709

cc: @DustinCampbell

### CODE REVIEW

File: src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs [hunk 1]
```diff
@@ -132,6 +132,7 @@ internal ReflectTypeDescriptionProvider()
             [typeof(ICollection)] = typeof(CollectionConverter),
             [typeof(Enum)] = typeof(EnumConverter),
             [s_intrinsicNullableKey] = typeof(NullableConverter),
+            [s_intrinsicReferenceKey] = typeof(ReferenceConverter),
```
Comment (ericstj) [inline src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs:135]: Tests?

Comment (safern) [reply src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs:135]: I thought we already had a test that tested this table but I see the InlineData is hardcoded, my first search was poor. Will add it, thanks 😄

Comment (ericstj) [reply src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs:135]: Sure.  Make sure you hit this codepath: https://github.com/dotnet/runtime/blob/5c274f699d9e63d5e3054dff0b995ec186b96426/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs#L1305-L1310

Comment (safern) [reply src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs:135]: Added.

**Review by ericstj — APPROVED** [review]

**Comment by safern** [comment]
WASM failures are: https://github.com/dotnet/runtime/issues/39473

**Pull request merged**